contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinUniswapV3.sol

```diff
@@ -39,19 +51,30 @@ contract MixinUniswapV3 {
         uint256 sellAmount,
         bytes memory bridgeData
     ) internal returns (uint256 boughtAmount) {
-        (IUniswapV3Router router, bytes memory path) = abi.decode(bridgeData, (IUniswapV3Router, bytes));
+        (address router, bytes memory path, uint256 routerVersion) = abi.decode(bridgeData, (address, bytes, uint256));
```

Comment:
Can we instead flat out replace router v1 with router v2 instead of encoding a version into the bridge data? Are there chains that don't have router v2 deployed?

Reply:
we can if we atomically update contract addresses and router address in `api` (could be problematic in the unlikely event of rollback)

this will require redeploying FQTs on all chains that uses MixinUniswapV3 😣

Reply:
ah good point